datadog/resource_datadog_synthetics_test_.go

```diff
@@ -466,24 +556,61 @@ func updateSyntheticsTestLocalState(d *schema.ResourceData, syntheticsTest *data
 	d.Set("request_headers", actualRequest.Headers)
 
 	actualAssertions := syntheticsTest.GetConfig().Assertions
-	var localAssertions []map[string]string
-	for _, assertion := range actualAssertions {
-		localAssertion := make(map[string]string)
-		if v, ok := assertion.GetOperatorOk(); ok {
-			localAssertion["operator"] = string(*v)
+	localAssertions := make([]map[string]interface{}, len(actualAssertions))
+	for i, assertion := range actualAssertions {
+		localAssertion := make(map[string]interface{})
+		if assertion.SyntheticsAssertionTarget != nil {
+			assertionTarget := assertion.SyntheticsAssertionTarget
+			if v, ok := assertionTarget.GetOperatorOk(); ok {
+				localAssertion["operator"] = string(*v)
+			}
+			if assertionTarget.HasProperty() {
+				localAssertion["property"] = assertionTarget.GetProperty()
+			}
+			if target := assertionTarget.GetTarget(); target != nil {
+				localAssertion["target"] = convertToString(target)
+			}
+			if v, ok := assertionTarget.GetTypeOk(); ok {
+				localAssertion["type"] = string(*v)
+			}
 		}
-		if assertion.HasProperty() {
-			localAssertion["property"] = assertion.GetProperty()
+		if assertion.SyntheticsAssertionJSONPathTarget != nil {
```

Comment:
Maybe a nit, but if these are mutually exclusive, would it make sense to make this an elif?